### PR TITLE
Auto-collapse sub-lists on initial list expansion

### PR DESCRIPTION
**Pull request by Shinsina**
Initial State:

Production:
Screenshot from 2022-11-14 13-56-51

Development:
Screenshot from 2022-11-14 13-56-54

"One level in":

Production:
Screenshot from 2022-11-14 13-57-02

Development:
Screenshot from 2022-11-15 14-18-59

Various additional expansions for Development:
Screenshot from 2022-11-15 14-19-30

### CODE REVIEW

**Comment by Shinsina** [comment]
[Collapse Directory Category subcategories if they have additional subcategories by default](https://trello.com/c/tLf7CACR/339-collapse-directory-category-subcategories-if-they-have-additional-subcategories-by-default)

File: services/app/app/templates/components/directory/category-tree.hbs [hunk 1]
```diff
@@ -1,18 +1,20 @@
 <div class={{ rootClass }}>
-  <label class="d-flex align-items-center">
-    {{#if node.children.length }}
-      {{entypo-icon "chevron-down" class="mr-2"}}
-    {{else}}
+  <label role="button" class="d-flex align-items-center" {{action "collapse"}}>
+    {{#if noChildren}}
```
Comment (solocommand) [inline services/app/app/templates/components/directory/category-tree.hbs]: You don't need to create a computed property for this, just use the `unless` control structure to negate an if -- `#unless node.children.length`.

File: services/app/app/templates/components/directory/category-tree.hbs [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
       <Input
         @type="checkbox"
         class="mr-2"
         @checked={{this.checked}}
         @change={{action "update"}}
       />
+    {{else if expandedState }}
+      {{entypo-icon "chevron-down" class="mr-2"}}
+    {{else if collapsedState }}
+      {{entypo-icon "chevron-right" class="mr-2"}}
```
Comment (solocommand) [inline services/app/app/templates/components/directory/category-tree.hbs]: No need for computed properties for this, just use an inline ternary -- 
```
{{ entypo collapsed ? "chevron-right" : "chevron-down" ... }}
```

File: services/app/app/templates/components/directory/category-tree.hbs [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
     {{/if}}
     {{ node.name }}
   </label>
-  {{#if node.children.length }}
+  {{#if expandedState }}
```
Comment (solocommand) [inline services/app/app/templates/components/directory/category-tree.hbs]: Remove extra prop

File: services/app/app/components/directory/category-tree.js [hunk 4]
```diff
@@ -1,9 +1,10 @@
 import Component from '@ember/component';
 import { computed } from '@ember/object';
+import ActionMixin from '@base-cms/company-update-app/mixins/action';
 
 const { error } = console;
 
-export default Component.extend({
+export default Component.extend(ActionMixin, {
```
Comment (solocommand) [inline services/app/app/components/directory/category-tree.js]: This component doesn't do any loading state manipulation -- why is this mixin being included?

File: services/app/app/components/directory/category-tree.js [hunk 5]
```diff
@@ -19,7 +20,28 @@ export default Component.extend({
     return ids.includes(id);
   }),
 
+  collapsed: computed('node.children', function() {
+    if (this.node.children.length) return true
+    return false
```
Comment (solocommand) [inline services/app/app/components/directory/category-tree.js]: Don't use (and overwrite) a computed property for this purpose. Instead, set the value for `collapsed` on init to a boolean based on `node.children.length`. Something like this:
```js
collapsed: true,
// ...
init() {
  this.collapsed = Boolean(getAsArray(node, 'children').length)
},
```

**Review by solocommand — CHANGES_REQUESTED** [review]

**Pull request merged**